meson.build

```diff
@@ -0,0 +1,455 @@
+armflags = []
+if host_cpu == 'arm'
+  arm_fpu = get_option('arm_fpu')
+  if arm_fpu == ''
+  elif arm_fpu == 'neon'
+    armflags = ['-mfpu=neon', '-mfloat-abi=hard']
+  elif arm_fpu == 'vfp-hard'
+    armflags = ['-mfpu=vfp', '-mfloat-abi=hard']
+  elif arm_fpu == 'vfp-soft'
+    armflags = ['-mfpu=vfp', '-mfloat-abi=softfp']
+  else
+    error('invalid arm-fpu option')
+  endif
+endif
+
+if host_machine.cpu().startswith('armv5') or host_machine.cpu().startswith('armv6')
+  cflags += '-DARMv5_ONLY'
+  message('Using ARMv5_ONLY')
+endif
+
+# Pass some compiler arguments
+foreach x : commonflags + warnflags + armflags
+  if cc.has_argument(x)
+    add_project_arguments(x, language: 'c')
+  endif
+endforeach
+foreach x : commonflags + warnflags + armflags + ['-fvisibility-inlines-hidden']
+  if cxx.has_argument(x)
+    add_project_arguments(x, language: 'cpp')
+  endif
+endforeach
+
+# If custom sharedir is specified add a define
+# otherwise use plateform specific sharedir
+sharedir = get_option('sharedir')
+if not (sharedir == '')
+  cflags += ['-DSHAREDIR=' + sharedir]
+else
+  # Also check that installation directory matches osal/files_{unix,win32}.c
+  if host_system == 'windows'
+    # Use the same location that installed binaries
+    sharedir = join_paths(get_option('prefix'), get_option('libdir'))
+  else
+    sharedir = join_paths(get_option('prefix'), get_option('datadir'), 'mupen64plus')
+  endif
+endif
+
+
+# declare required dependencies
+zlib = dependency('zlib')
+libpng = dependency('libpng')
+# try sdl2 first, and fallback to sdl1 otherwise
+sdl = dependency('sdl2', required : false)
+if not sdl.found()
+  sdl = dependency('sdl')
+endif
+# Add math library portably (see meson FAQ)
+libm = cc.find_library('m', required : false)
+# Use minizip when available, use embedded version otherwise
+# XXX: use the dependency fallback mecanism ?
+minizip = dependency('minizip', required : false)
+if minizip.found()
+  cflags += ['-DLIBMINIZIP']
+  extra_dep += minizip
+else
+  src += [
+      'src/main/zip/ioapi.c',
+      'src/main/zip/zip.c',
+      'src/main/zip/unzip.c']
+  cflags += ['-DNOCRYPT', '-DNOUNCRYPT']
+endif
+# Apparently FreeBSD needs this
+if host_system == 'bsd'
+  cflags += ['-DIOAPI_NO_64']
+endif
+
+if host_system == 'mingw'
+  pthreads = dependency('threads')
+  extra_dep += pthreads
+endif
+
+
+# Add OSAL sources
+if host_system == 'windows'
+  src += [
+    'src/osal/dynamiclib_win32.c',
+    'src/osal/files_win32.c']
+else
+  src += [
+    'src/osal/dynamiclib_unix.c',
+    'src/osal/files_unix.c']
+  # On unix platforms we also need libdl
+  libdl = cc.find_library('dl')
+  extra_dep += libdl
+endif
+
+
+# Add various core debuging possibilities
+if with_dbg_compare
+  cflags += ['-DCOMPARE_CORE']
+endif
+if with_dbg_core
+  cflags += ['-DCORE_DBG']
+endif
+if with_dbg_count
+  cflags += ['-DCOUNT_INSTR']
+endif
+if with_dbg_timing
+  cflags += ['-DPROFILE']
+  extra_dep += cc.find_library('rt')
```

Comment:
OpenBSD keeps librt entirely in libc, so there is no librt to find. It’s the odd one out in this regard, so this should be wrapped in “if not openbsd”.